back/app/services/survey_results_generator_service.rb

```diff
@@ -23,6 +23,16 @@ def generate_results
     }
   end
 
+  def generate_question_result(field_id)
+    field = fields.find { |f| f.id == field_id }
+    return unless field
```

Comment:
Here I was wondering if it's ok to return like this? Should I throw an error or something instead?